### DISCUSSION TITLE
Creating symlinks for a binary

### DISCUSSION THREAD

**Discussion by ducaale** [Homebrew/discussions#963] [Everyday usage] [answered]
We have recently added a feature in https://github.com/ducaale/xh where the binary will behave differently depending on the name. For example:

```sh
# make a GET request to http://example.org
$ xh example.org

# make a GET request to https://example.org
$ xhs example.org
```

Is it possible to symlink/copy `xh` binary after it is installed to support this? Thanks!

**Comment by carlocab** [comment, marked as answer]
I think you can use the [`install_symlink`](https://rubydoc.brew.sh/Pathname.html#install_symlink-instance_method) method for this.

**Reply by ducaale** [reply]
Would it be something like this?
```rb
def install
  system "cargo", "install", *std_cargo_args
  bin.install_symlink "xhs"
end
```

**Reply by carlocab** [reply]
Not quite, I think it needs to be
```ruby
bin.install_symlink bin/"xh" => "xhs"
```
Otherwise, you haven't specified what the target of your symlink is.

**Reply by ducaale** [reply]
Got it. Thanks!
